File: updater/src/main/java/identifiers/ByteArrayPool.java
package identifiers;

import org.objectweb.asm.Type;
import utility.AbstractIdentifier;
import utility.ClassWrapper;
import utility.TypeUtilities;

public class ByteArrayPool extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.getInstanceFields().isEmpty() &&
                classNode.getInstanceMethods().isEmpty() &&
                classNode.getCountStaticFieldsOfType(TypeUtilities.withDimensions(Type.BYTE_TYPE, 2)) >= 3;
    }
}

File: find-names/src/main/java/mapper/identifiers/classes/ByteArrayPool.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.withDimensions
import mapper.wrappers.ClassWrapper
import org.objectweb.asm.Type.BYTE_TYPE

class ByteArrayPool : IdentityMapper.Class() {

    override val predicate = predicateOf<ClassWrapper> { it.instanceFields.isEmpty() }
            .and { it.instanceMethods.isEmpty() }
            .and { it.staticFields.count { it.type == BYTE_TYPE.withDimensions(2) } >= 3}
}